torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1799,12 +1800,23 @@ def hint():
                 track_symint(source, t)
                 continue
             assert isinstance(t, torch.Tensor)
-            for i, ss in enumerate(t.size()):
-                property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
-                track_symint(property_source, ss, constraint[i])
-            for i, ss in enumerate(t.stride()):
-                track_symint(TensorPropertySource(source, TensorProperty.STRIDE, i), ss)
-            track_symint(TensorPropertySource(source, TensorProperty.STORAGE_OFFSET), t.storage_offset())
+            if t.is_nested:
+                assert all([c is None for c in constraint]), \
+                    "Dim constraints are not supported for nested tensors"
+                track_symint(NestedTensorPropertySource(source, TensorProperty.SIZE),
+                             t._nested_tensor_size().size(0))
+                track_symint(NestedTensorPropertySource(source, TensorProperty.STRIDE),
+                             t._nested_tensor_strides().size(0))
+                track_symint(NestedTensorPropertySource(source, TensorProperty.STORAGE_OFFSET),
+                             t._nested_tensor_storage_offsets().size(0))
```

Comment:
@ezyang does this match what you expect here? I'm scared we're not guarding on enough because the metadata is so symbolic.

Reply:
Hmm, well, it's better. But the way to figure out if you're guarding enough, is to just exhaustively run track_symint on every internal data structure.  I don't agree with the NestedTensorPropertySource structure by the way; you should make an entirely new enum for stuff like `_nested_tensor_size`, and then you can index each entry individually

Reply:
Okay gotcha, I changed it work recursively on the metadata tensors. Now it's more aggressively guarding on size, stride, and storage offset of `_nested_tensor_size`, `_nested_tensor_strides`, and `_nested_tensor_storage_offsets`, but I'm not convinced this buys us much in practice.

Since, for example, contiguity of a nested tensor is defined by the *values* of `_nested_tensor_size`, `_nested_tensor_strides`, and `_nested_tensor_storage_offsets`, is it still possible we're not guarding enough? Like we are guarding on the metadata structure e.g. `(s0, n)`, but this structure is fairly permissive. We can pass those checks and still not know if the NT is contiguous.